pkg/base/registry/nacos/registry.go

```diff
@@ -0,0 +1,174 @@
+package nacos
+
+import (
+	"encoding/json"
+	"errors"
+	"log"
+	"net"
+	"strconv"
+	"strings"
+)
+import (
+	"github.com/nacos-group/nacos-sdk-go/clients"
+	"github.com/nacos-group/nacos-sdk-go/clients/naming_client"
+	"github.com/nacos-group/nacos-sdk-go/model"
+	"github.com/nacos-group/nacos-sdk-go/vo"
+)
+import (
+	nacosConstant "github.com/nacos-group/nacos-sdk-go/common/constant"
+	"github.com/transaction-wg/seata-golang/pkg/base/common/constant"
+	"github.com/transaction-wg/seata-golang/pkg/base/common/extension"
+	"github.com/transaction-wg/seata-golang/pkg/base/registry"
+	clientConfig "github.com/transaction-wg/seata-golang/pkg/client/config"
+	"github.com/transaction-wg/seata-golang/pkg/tc/config"
+)
+
+func init() {
+	extension.SetRegistry(constant.NACOS_KEY, newNacosRegistry)
+}
+
+type nacosRegistry struct {
+	namingClient naming_client.INamingClient
+}
+type nacosEventListener struct {
+}
+
+func (nr *nacosEventListener) OnEvent(service []*registry.Service) error {
+	data, err := json.Marshal(service)
+
+	log.Print("servie info change：" + string(data))
```

Comment:
for logs in this pr, I think it's better use log from  https://github.com/opentrx/seata-golang/blob/dev/pkg/util/log/logging.go.

Reply:
understand